src/Ext/House/Body.cpp

```diff
@@ -622,3 +622,73 @@ DEFINE_HOOK(0x50114D, HouseClass_InitFromINI, 0x5)
 
 	return 0;
 }
+
+int HouseExt::CountOwnedIncludeDeploy(const HouseClass* pThis, const TechnoTypeClass* pItem)
+{
+	int count = pThis->CountOwnedNow(pItem);
+	count += pItem->DeploysInto == nullptr ? 0 : pThis->CountOwnedNow(pItem->DeploysInto);
+	count += pItem->UndeploysInto == nullptr ? 0 : pThis->CountOwnedNow(pItem->UndeploysInto);
+	return count;
+}
+
+CanBuildResult HouseExt::BuildLimitGroupCheck(const HouseClass* pThis, const TechnoTypeClass* pItem, bool buildLimitOnly, bool includeQueued)
+{
+	const auto pItemExt = TechnoTypeExt::ExtMap.Find(pItem);
+
+	if (pItemExt->BuildLimitGroup_Types.empty())
+		return CanBuildResult::Buildable;
+
+	if (pItemExt->BuildLimitGroup_ContentIfAnyMatch.Get())
+	{
+		bool reachedLimit = false;
+
+		for (size_t i = 0; i < std::min(pItemExt->BuildLimitGroup_Types.size(), pItemExt->BuildLimitGroup_Nums.size()); i++)
+		{
+			TechnoTypeClass* pType = pItemExt->BuildLimitGroup_Types[i];
+			int ownedNow = CountOwnedIncludeDeploy(pThis, pType);
```

Comment:
```suggestion
			int ownedNow = HouseExt::CountOwnedIncludeDeploy(pThis, pType);
```
same to other references